include/silo/common/nucleotide_symbols.h

```diff
@@ -125,6 +129,24 @@ inline std::optional<NUCLEOTIDE_SYMBOL> charToNucleotideSymbol(char character) {
          return std::nullopt;
    }
 }
+
+inline size_t toNucleotideSymbolId(char c) {
```

Comment:
Functions like these were once in the codebase but I would like to keep them out and use the enum classes NucleotideSymbol and AASymbol.
For the indexing reasons I think you preferred these functions, you can have a look at NucleotideSymbolMap, which also uses a constexpr sized array for storage and lookup of value types for a nucleotide symbol